README update and CONTRIBUTING add (#68)
gustavo-depaula/stalin-sort: CONTRIBUTING.md

## 2. Start coding! :rabbit2:

Once you have forked the repo, add your program in the language folder in main branch, if there is no language folder, make one, then add into it.

For a PR to be approved and merged please note:

- We allow more than one solution for one language, so if you don't want to modify someone else's code, you can add a new file with your own solution. Have in mind that your solution must be completely unique, if it isn't consider modifying the existing code.
- We don't allow deletions of others files or code.
- We don't allow files to be directly added in root directory of project, add them in their language folder
- Follow the code specifications!

### Code specifications

For the purpuses of this repo, we defined some rules that your code must to follow, this rules are listed bellow:
1. Your file **must** be named **stalin-sort**. All the other files of the same language must be at its title the implementation expecificaiton. For example the C++ folder have the *stalin-sort.cpp* and the *stalin-sort-compile-time.cpp* that runs at compile time, just as said at its name. 

> 2. Your file **must** have only **ONE** function called Stalin Sort, for pattern matters it can be named according to the language case, so if it's PascalCase the function name is StalinSort, if it's camelCase the function name is stalinSort and so on. Take note that the language case must be followed through all code. Don't put the main function, or any kind of tests at the file.

Reviewer: main function and tests help us in running the code ourselves
```suggestion
2. Your file **must** have only **ONE** function called Stalin Sort, for pattern matters it can be named according to the language case, so if it's PascalCase the function name is StalinSort, if it's camelCase the function name is stalinSort and so on. Take note that the language case must be followed through all code.
```

> 3. For this repo, we're implementing the ordering of **INTEGERS**, so case the language that you're implementing is static typed have it in mind.

Reviewer: ```suggestion
3. For this repo, we're implementing the ordering of **INTEGERS**, so case the language that you're implementing is static typed have it in mind. Additionally, you could implement the ordering of other types, but the focus here is on integers.
```

# CONTRIBUTING

> We're accepiting help from anyone! This repo is collaborative, so if you want to add a solution in a new language (check Wikipedia [Programming Language List](https://en.wikipedia.org/wiki/List_of_programming_languages)) or just improve some code that you think that can be improved, follow this steps:

Reviewer 2: ```suggestion
We're accepting help from anyone! This repo is collaborative, so if you want to add a solution in a new language (check Wikipedia [Programming Language List](https://en.wikipedia.org/wiki/List_of_programming_languages)) or just improve some code that you think that can be improved, follow this steps:
```

## 1. Fork it :fork_and_knife:

You can get your own fork/copy of [Stalin Sort](https://github.com/gustavo-depaula/stalin-sort) by using the <a  href="https://github.com/Hacktoberfest-2018/Hello-world/new/master?readme=1#fork-destination-box"><kbd><b>Fork</b></kbd></a> button or clicking [this](https://github.com/gustavo-depaula/stalin-sort/new/master?readme=1#fork-destination-box).

  [![Fork Button](https://help.github.com/assets/images/help/repository/fork_button.jpg)](https://github.com/gustavo-depaula/stalin-sort/new/master?readme=1#fork-destination-box)

## 2. Start coding! :rabbit2:

Once you have forked the repo, add your program in the language folder in main branch, if there is no language folder, make one, then add into it.

For a PR to be approved and merged please note:

> - We allow more than one solution for one language, so if you don't want to modify someones code, you can add a new file with your own solution. Have in mind that your solution must be completely unique, if it isn't consider modify the existing code.

Reviewer 2: ```suggestion
- We allow more than one solution for one language, so if you don't want to modify someone else's code, you can add a new file with your own solution. Have in mind that your solution must be completely unique, if it isn't consider modifying the existing code.
```

> - We allow maximum of 5000 additions (your PR will get close if its exceeds, but you can ask maintainer for reviewing this)

Reviewer 2: ```suggestion

```

- We don't allow deletions of others files or code.

> - We don't allow files to be directly added in root directory of project, add them in there language folder

Reviewer 2: ```suggestion
- We don't allow files to be directly added in root directory of project, add them in their language folder
```

- Follow the code specifications!

### Code specifications

For the purpuses of this repo, we defined some rules that your code must to follow, this rules are listed bellow:

> 1. Your file **must** be named **stalin-sort**. All the other files of the same language must follow a numeric order, so the second implementation **must** be named as **stalin-sort-2**.

Reviewer: this is not helpful. if we have more than one version of the same language, is better to have it named like "stalin-sort-compile-time" so we can know what it does from the name

> 2. All the multiple implementations **must** starts with a commentary at its top explaining in what this implementation differs from the first one. For example the C++ *stalin-sort-2.cpp* has a commentary at its top saying that this one is a compile time implementation.

Reviewer: too much bureaucracy

> 3. Your file **must** have only **ONE** function called Stalin Sort, for pattern matters it can be named according to the language case, so if it's PascalCase the function name is StalinSort, if it's camelCase the function name is stalinSort and so on. Take note that the language case must be followed through all code.

Reviewer: too much bureaucracy

4. For this repo, we're implementing the ordering of **INTEGERS**, so case the language that you're implementing is static typed have it in mind.

> 5. Your function **must** receive only **ONE** argument, an array or a pointer, and **must** return a **NEW** array or pointer, according to the receved one, the received one shouldn't be modified.

Reviewer: some languages don't even have functions, so this is not helpful either

> 6. Your file **must** at its end export the function. Case the language works like Elixir and doesn't need to explicitly exports the function, add a commentary at the end of the file saying it.

Reviewer: too much bureaucracy
